### PR TITLE
3dfloodfill selection

### PR DESCRIPTION
**Pull request by PennyQ**
The subset doesn't get created :(

### CODE REVIEW

File: glue_exp/tools/floodfill_selection/floodfill_selection.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    # change to a MaskSubsetState.
+    subset_state = PatchedElementSubsetState(visible_data, np.where(mask)[0])
+
+    # We now check what the selection mode is, and update the selection as
+    # needed (this is delegated to the correct subset mode).
+    mode = EditSubsetMode()
+    focus = visible_data[0] if len(visible_data) > 0 else None
+    # mode.update(self._data_collection, subset_state, focus_data=focus)
+    print('input for mark_selected', data_object, subset_state, focus)
+    mode.update(data_object, subset_state, focus_data=focus)
```
Comment (PennyQ) [inline glue_exp/tools/floodfill_selection/floodfill_selection.py:187]: @astrofrog  I don't know how to get current data_collection so I use the `imageWidget.client.display_data` as the first input for `mode.update()`, is it the reason why there is no subset showed up? How does the `mode.update` find current subset and update it?

File: glue_exp/tools/floodfill_selection/floodfill_selection.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    if mask is not None:
+        new_mask = np.reshape(mask, data.shape)
+        new_mask = np.ravel(np.transpose(new_mask))
+        print('new_mask', np.sum(new_mask))
+        mark_selected(new_mask, data, data_object)
+    # if xy is None:
+    #     return None
+
+    # p = roi.PolygonalROI(vx=xy[:, 0], vy=xy[:, 1])
+    # return p
```
Comment (PennyQ) [inline glue_exp/tools/floodfill_selection/floodfill_selection.py:172]: Is a subset get created when defining a `roi` as in commented line? Not clear about the relation between `roi` and subset :(

File: glue_exp/tools/floodfill_selection/floodfill_selection.py (hunk 1 above)
Comment (PennyQ) [inline glue_exp/tools/floodfill_selection/floodfill_selection.py:187]: Another question, why in `iPython` we could easily get current data_collection with  typing `dc`? But I didn't find how to get it from `ImageWidget` or any related classes?

**Comment by astrofrog** [comment]
@PennyQ - I'll look at this today

**Pull request closed without merging**